### PR TITLE
[After #443] Feature: add support for filtering twingate groups datasource

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves https://github.com/Twingate/terraform-provider-twingate/issues/432

Tests passes: https://github.com/vmanilo/terraform-provider-twingate/actions/runs/7676594341

## Changes
- added optional filters for groups datasource

### CODE REVIEW

File: docs/data-sources/groups.md [hunk 1]
```diff
@@ -28,8 +33,13 @@ data "twingate_groups" "foo" {
 
 - `groups` (Attributes List) List of Groups (see [below for nested schema](#nestedatt--groups))
```
Comment (twingate-blee) [inline docs/data-sources/groups.md:34]: Is there a difference between

```
data "twingate_groups" "foo" {
  groups = {
    name = "test"
    is_active = true
  }
}  
``` 

and

```
data "twingate_groups" "foo" {
  name = "test"
  is_active = true
}  
```

Do we still need the `groups` attribute? If so should we just add the filters to it?

Comment (ekampf) [reply docs/data-sources/groups.md:34]: We dont have this extra hierarchy in other datasources?
Personally, I would rather have a breaking change and get rid of it  than have this weird duplication of functionality

Comment (ekampf) [reply docs/data-sources/groups.md:34]: I see we have this pattern in all our datasources...

Comment (vmanilo) [reply docs/data-sources/groups.md]: fixed

**Review by twingate-blee — APPROVED** [review]
lgtm

**Pull request merged**